internal/controllers/addon/controller.go

```diff
@@ -208,13 +201,11 @@ func (r *AddonReconciler) SetupWithManager(mgr ctrl.Manager) error {
 func (r *AddonReconciler) Reconcile(
 	ctx context.Context, req ctrl.Request) (res ctrl.Result, err error) {
 
-	// inject namespace and name into context for the loggers within the sub-reconcilers
-	// to pick it up.
-	ctx = context.WithValue(ctx, addonNamespaceNameKey, req.NamespacedName.String())
 	log := r.Log.WithValues("addon", req.NamespacedName.String())
+	ctx = controllers.ContextWithLogger(ctx, log)
 
 	addon := &addonsv1alpha1.Addon{}
-	if err := r.Get(ctx, req.NamespacedName, addon); err != nil {
+	if err := r.Get(controllers.ContextWithLogger(ctx, log), req.NamespacedName, addon); err != nil {
```

Comment:
Isn't this redundant, since we're already setting a logger in the context from the line right above? Should it just be `r.Get(ctx, req.NamespacedName, addon)`? I'm not sure I understand why we have to set a logger here again 🤔